src/nccl_ofi_rdma.c

```diff
@@ -1450,70 +1515,76 @@ static inline int process_completions(struct fi_cq_tagged_entry *cq_entry,
 	return ret;
 }
 
-static const char *req_state_str(nccl_net_ofi_rdma_req_state_t state)
+/*
+ * @brief	Process error completion entries from the CQ error queue
+ *
+ * @return	0, on success
+ *		error, on others
+ */
+static inline int process_err_completion(nccl_net_ofi_rdma_ep_t *ep,
+					 int rail_id)
 {
-	switch(state) {
-	case NCCL_OFI_RDMA_REQ_CREATED:
-		return "CREATED";
-	case NCCL_OFI_RDMA_REQ_PENDING:
-		return "PENDING";
-	case NCCL_OFI_RDMA_REQ_COMPLETED:
-		return "COMPLETED";
-	case NCCL_OFI_RDMA_REQ_ERROR:
-		return "ERROR";
-	}
-	return "unknown";
-}
+	nccl_net_ofi_ep_rail_t *rail = get_rail(ep, rail_id);
+	struct fi_cq_err_entry err_entry = { 0 };
+	nccl_net_ofi_rdma_req_t *req = NULL;
+	int ret = 0;
+	int rc = 0;
 
-static const char *req_type_str(nccl_net_ofi_rdma_req_type_t type)
-{
-	switch(type) {
-	case NCCL_OFI_RDMA_SEND_CONN:
-		return "SEND_CONN";
-	case NCCL_OFI_RDMA_SEND_CONN_RESP:
-		return "SEND_CONN_RESP";
-	case NCCL_OFI_RDMA_RECV_CONN:
-		return "RECV_CONN";
-	case NCCL_OFI_RDMA_RECV_CONN_RESP:
-		return "RECV_CONN_RESP";
-	case NCCL_OFI_RDMA_SEND:
-		return "SEND";
-	case NCCL_OFI_RDMA_RECV:
-		return "RECV";
-	case NCCL_OFI_RDMA_SEND_CTRL:
-		return "SEND_CTRL";
-	case NCCL_OFI_RDMA_RECV_SEGMS:
-		return "RECV_SEGMS";
-	case NCCL_OFI_RDMA_BOUNCE:
-		return "BOUNCE";
-	case NCCL_OFI_RDMA_FLUSH:
-		return "FLUSH";
-	case NCCL_OFI_RDMA_EAGER_COPY:
-		return "EAGER_COPY";
+	rc = fi_cq_readerr(rail->cq, &err_entry, 0);
+	if (OFI_UNLIKELY(rc == -FI_EAGAIN)) {
+		/*
+		 * Error not available yet.
+		 * fi_cq_read will keep returning -FI_EAVAIL so just bail out and try again later.
+		 */
+		return ncclSuccess;
+	} else if (OFI_UNLIKELY(rc < 0)) {
+		NCCL_OFI_WARN("Unable to read from fi_cq_readerr. RC: %zd. Error: %s",
```

Comment:
rc is an int; the format should be `%d`, not `%zd`.  I'm also not sure why you have both an rc and ret here.

Reply:
Kept the existing log and rc/ret separation from the original function, will gladly change it.